indico/modules/events/registration/controllers/api/checkin.py

```diff
@@ -130,3 +132,28 @@ def _process_args(self):
 
     def _process_GET(self):
         return CheckinRegistrationSchema().jsonify(self.registration)
+
+
+@json_errors
+@cors
+@oauth_scope('registrants')
+class RHCheckinAPIRegistrationCustomQRCode(RH):
+    """Get indico style ticket details for a specific registration from a custom plugin QR Code."""
+
+    @use_kwargs({
+        'data': fields.String(required=True),
+        'qr_name': fields.String(required=True),
+    }, location='json')
+    def _process_POST(self, data, qr_name):
+        try:
+            custom_qr_code_handler = get_custom_ticket_qr_code_handlers()[qr_name]
+        except KeyError:
+            return
+        if reg := custom_qr_code_handler.lookup_registration(data):
+            url = config.BASE_URL.removeprefix('https://')
+            qr_code_version = 2
+            result = {
+                'i': [qr_code_version, url, _base64_encode_uuid(reg.ticket_uuid)]
+            }
+            return jsonify(result)
+        return
```

Comment:
```suggestion
        raise NotFound('No registration found')
```

and you can bail out early, ie `if not (reg := ...): raise ...` instead of having everything else inside the `if` block